src/app/(main)/AndroidUpgradeNotification.tsx

```diff
@@ -0,0 +1,41 @@
+'use client';
+
+import { useEffect, useState } from 'react';
+import Flex from '../../components/Flex';
+
+export default function AndroidUpgradeNotification() {
+  const [alertVisible, setAlertVisible] = useState(true);
+  const androidUpgradeNotificationDismissed = localStorage.getItem(
```

Comment:
is this synchronous and okay to use in the render function? if so, just pass `!androidUpgradeNotificationDismissed` into the `useState` default and you can drop the `useEffect` below :)

Reply:
this may break because of SSR. if so you can make this component imported dynamically so it only renders on the client

```
import dynamic from 'next/dynamic'
 
const DynamicHeader = dynamic(() => import('../components/header'), {
  ssr: false,
})
```